### PR TITLE
EVG-6492: redact GitHub key from StdErr

### PR DESCRIPTION
**Pull request by ybrill**
Copied from https://github.com/evergreen-ci/evergreen/blob/ece140675f3759f3829e34d37dd14633be8ef045/command/git.go#L337

### CODE REVIEW

File: command/git_push.go [hunk 1]
```diff
@@ -24,6 +25,7 @@ type gitPush struct {
 	DryRun         bool   `yaml:"dry_run" mapstructure:"dry_run"`
 	CommitterName  string `yaml:"committer_name" mapstructure:"committer_name"`
 	CommitterEmail string `yaml:"committer_email" mapstructure:"committer_email"`
+	Token          string `plugin:"expand" mapstructure:"token"`
```
Comment (john-m-liu) [inline command/git_push.go]: Explicitly set a yaml tag for this

File: command/git_push_test.go [hunk 2]
```diff
@@ -75,6 +87,13 @@ func TestGitPush(t *testing.T) {
 		},
 		"PushPatch": func(*testing.T) {
 			manager := &jasper.MockManager{}
+			manager.Create = func(opts *jasper.CreateOptions) jasper.MockProcess {
+				_, err = opts.Output.Error.Write([]byte(fmt.Sprintf("The key: %s", token)))
```
Comment (john-m-liu) [inline command/git_push_test.go]: This will end up writing the message twice won't it?

Comment (ybrill) [reply command/git_push_test.go]: I think three times.
I changed the test to just check the last message.

File: command/git_push.go (hunk 1 above)
Comment (ybrill) [reply command/git_push.go]: done.
Why is this important? Would someone put their token as a parameter in their config file?

Comment (john-m-liu) [reply command/git_push.go]: Isn't that why you added this parameter? To allow users to override the global token if needed?

Comment (ybrill) [reply command/git_push.go]: my thought was if a token were a project variable it would get expanded and override the global token.

Comment (john-m-liu) [reply command/git_push.go]: That was my thought too. How would it be passed in if not in the config file?

Comment (ybrill) [reply command/git_push.go]: I get it now (@Kimchelly explained it to me). The token is passed as a parameter in the config file, but could just be an expansion of a project level variable. 
I had previously been envisioning a project level variable called "token" that ends up in the task's expansions.

**Pull request merged**